forui/lib/src/widgets/label.dart

```diff
@@ -0,0 +1,444 @@
+            child: DefaultTextStyle(
+              style: stateStyle.descriptionTextStyle,
+              textHeightBehavior: const TextHeightBehavior(applyHeightToFirstAscent: false),
+              child: description!,
+            ),
+          ),
+        if (error != null && state == FLabelState.error)
+          Padding(
+            padding: style.errorPadding,
+            child: DefaultTextStyle(
+              style: style.formFieldStyle.errorStyle.errorTextStyle,
+              textHeightBehavior: const TextHeightBehavior(applyHeightToFirstAscent: false),
+              child: error!,
+            ),
+          ),
+      ],
+    );
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty('style', style))
+      ..add(EnumProperty('state', state));
+  }
+}
+
+/// The [FLabel]'s styles.
+final class FLabelStyles with Diagnosticable {
+  /// The horizontal label's style.
+  final FLabelStyle horizontal;
+
+  /// The vertical label's style.
+  final FLabelStyle vertical;
+
+  /// Creates a [FLabelStyles].
+  const FLabelStyles({
+    required this.horizontal,
+    required this.vertical,
+  });
+
+  /// Creates a [FLabelStyles] that inherits its properties from the given [style].
+  FLabelStyles.inherit({required FStyle style})
+      : horizontal = FLabelStyle(
+          formFieldStyle: style.formFieldStyle,
+          childPadding: const EdgeInsets.only(right: 8),
+          descriptionPadding: const EdgeInsets.only(top: 2),
+          errorPadding: const EdgeInsets.only(top: 2),
+        ),
+        vertical = FLabelStyle(
+          formFieldStyle: style.formFieldStyle,
+          labelPadding: const EdgeInsets.only(bottom: 5),
+          descriptionPadding: const EdgeInsets.only(top: 5),
+          errorPadding: const EdgeInsets.only(top: 5),
+        );
+
+  /// Returns a copy of this [FLabelStyles] with the given properties replaced.
+  @useResult
+  FLabelStyles copyWith({
+    FLabelStyle? horizontal,
+    FLabelStyle? vertical,
+  }) =>
+      FLabelStyles(
+        horizontal: horizontal ?? this.horizontal,
+        vertical: vertical ?? this.vertical,
+      );
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty('horizontal', horizontal))
+      ..add(DiagnosticsProperty('vertical', vertical));
+  }
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is FLabelStyles &&
+          runtimeType == other.runtimeType &&
+          horizontal == other.horizontal &&
+          vertical == other.vertical;
+
+  @override
+  int get hashCode => horizontal.hashCode ^ vertical.hashCode;
+}
+
+/// The [FLabel]'s style.
+final class FLabelStyle with Diagnosticable {
+  /// The form field's style.
+  final FFormFieldStyle formFieldStyle;
```

Comment:
I think we should remove the `formFieldStyle` parameter from `FStyle.copyWith(...)` method.

Reply:
Why though?

Reply:
FLabelStyle depends on `FFormFieldStyle`. Updating FStyle`s `FFormFieldStyle` doesn't cascade to FLabelStyle.

Reply:
Doesn't the issue exist for all the other parameters as well? It could be a better idea to get rid of `copyWith` for `style`, `colorScheme`, and `typography` if that's the concern.

Reply:
It seems like I was thinking of something else, https://github.com/forus-labs/forui/pull/132/files